chainer/training/evaluator.py

```diff
@@ -0,0 +1,195 @@
+
+from chainer.dataset import convert
+from chainer.dataset import iterator as iterator_module
+
+
+class Evaluator(object):
+    """Base class of all evaluators."""
+
+    def run(self):
+        raise NotImplementedError
+
+    def evaluate(self):
+        raise NotImplementedError
+
+    def get_iterator(self, name):
+        raise NotImplementedError
+
+    def get_all_iterators(self):
+        raise NotImplementedError
+
+    def get_target(self, name):
+        raise NotImplementedError
+
+    def get_all_targets(self):
+        raise NotImplementedError
+
+
+def eval_func_with_volatile(eval_func, in_arrays):
+    if isinstance(in_arrays, tuple):
+        in_vars = tuple(variable.Variable(x, volatile='on') for x in in_arrays)
+        eval_func(*in_vars)
+    elif isinstance(in_arrays, dict):
+        in_vars = {
+            key: variable.Variable(x, volatile='on')
+            for key, x in six.iteritems(in_arrays)
+        }
+        eval_func(**in_vars)
+    else:
+        in_var = variable.Variable(in_arrays, volatile='on')
+        eval_func(in_var)
+
+
+class StandardEvaluator(Evaluator):
+    """Standard implementation of Evaluator.
+
+    This is the standard implementation of :class:`Evaluator`.
+    It evaluate *target link* with given iterator and summarize the reported
+    values in the link by calling :func:`~chainer.reporter.report`.
+    It accepts one or more iterators and one or more links.
+    The default evaluate routine assumes that there is only one training
+    dataset and one model. Users can override this evaluate routine by
+    inheriting this class and overriding the :meth:`evaluate` method. Each
+    batch is converted to input arrays by
+    :func:`~chainer.datasets.concat_examples` by default,
+    which can also manually set by ``converter`` argument.
+
+        >>> from chainer.training.evaluator import StandardEvaluator
+        >>> import chainer
+        >>> import numpy as np
+        >>>
+        >>> model = chainer.links.Classifier(
+        >>>         chainer.links.Linear(
+        >>>                 2, 2,
+        >>>                 nobias=True,
+        >>>                 initialW=np.identity(2, dtype=np.float32),
+        >>>         )
+        >>> )
+        >>>
+        >>> data = [
+        >>>     (np.asarray([1, 0], dtype=np.float32), np.int32(0)),  # true
+        >>>     (np.asarray([1, 0], dtype=np.float32), np.int32(0)),  # true
+        >>>     (np.asarray([0, 1], dtype=np.float32), np.int32(1)),  # true
+        >>>     (np.asarray([0, 1], dtype=np.float32), np.int32(1)),  # true
+        >>>     (np.asarray([0, 1], dtype=np.float32), np.int32(0)),  # false
+        >>> ]
+        >>>
+        >>> iterator = chainer.iterators.SerialIterator(data, 5, repeat=False)
+        >>>
+        >>> evaluator = StandardEvaluator(iterator, model)
+        >>> result = evaluator.run()
+        >>> print(result)
+        {'main/accuracy': 0.8000000119209289, 'main/loss': 0.5132616162300109}
```

Comment:
I think this example is useful for testing `StandardEvaluator`. So, could you write unittests from it?